src/lib/list/list.ts

```diff
@@ -6,28 +6,54 @@ import {
     QueryList,
     Directive,
     ElementRef,
+    Inject,
+    Input,
+    OpaqueToken,
     Renderer,
     AfterContentInit,
     NgModule,
     ModuleWithProviders,
 } from '@angular/core';
-import {MdLine, MdLineSetter, MdLineModule} from '../core';
+import {MdLine, MdLineSetter, MdLineModule, MdRippleModule} from '../core';
 
 @Directive({
   selector: 'md-divider'
 })
 export class MdListDivider {}
 
+export enum ListType {
+  NORMAL_LIST,
+  NAV_LIST,
+}
+
+/**
+ * Token used to inject the list type into child MdListItem components so they can know whether
+ * they're in a nav list (and thus should use an MdRipple).
+ */
+const LIST_TYPE_TOKEN = new OpaqueToken('is_nav_list');
+
 @Component({
   moduleId: module.id,
-  selector: 'md-list, md-nav-list',
+  selector: 'md-list',
   host: {'role': 'list'},
   template: '<ng-content></ng-content>',
   styleUrls: ['list.css'],
+  providers: [{ provide: LIST_TYPE_TOKEN, useValue: ListType.NORMAL_LIST }],
   encapsulation: ViewEncapsulation.None
 })
 export class MdList {}
 
+@Component({
+  moduleId: module.id,
+  selector: 'md-nav-list',
+  host: {'role': 'list'},
+  template: '<ng-content></ng-content>',
+  styleUrls: ['list.css'],
+  providers: [{ provide: LIST_TYPE_TOKEN, useValue: ListType.NAV_LIST }],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdNavList {}
```

Comment:
Rather than creating a new component for `md-nav-list`, you can create an additional `@Directive` that also matches `md-nav-list` with a `providers` entry.